src/buttons/button-icon.spec.ts

```diff
@@ -0,0 +1,57 @@
+import {it, describe, expect, injectAsync, TestComponentBuilder} from 'angular2/testing';
+import {Component} from 'angular2/core';
+import {NglButtonIcon} from './button-icon';
+
+function getButtonElement(element: Element): HTMLButtonElement {
+  return <HTMLButtonElement>element.querySelector('button');
+}
+
+describe('`nglButtonIcon`', () => {
+
+  it('should render the default button icon when the type is empty', testAsync(
+    `<button [nglButtonIcon]="style">Go <ngl-icon-button icon="right"></ngl-icon-button></button>`, ({fixture, done}) => {
```

Comment:
@esarbanis No need to have `<ngl-icon-button icon="right"></ngl-icon-button>` in your HTML, as we don't test this.